docs/literate/sketches/preorders.jl

```diff
@@ -0,0 +1,270 @@
+# # Preorders
+#
+#md # [![](https://img.shields.io/badge/show-nbviewer-579ACA.svg)](@__NBVIEWER_ROOT_URL__/generated/sketches/Preorders.ipynb)
+# Many of the ideas in Category Theory can be viewed as generalizations of 
+# preorders or monoids. This sketch shows some features of Catlab through the 
+# lens of preorders.
+# You will see examples of defining GATs, Presentations, Syntax, and Functors.
+# These are illustrated with preorders or thin categories, which are particularly 
+# simple cases of categories. 
+
+using Core: GeneratedFunctionStub
+using Catlab
+using Catlab.Theories
+using Catlab.Programs
+using Catlab.WiringDiagrams
+using Catlab.Graphics
+using Catlab.Graphics.Graphviz
+using Catlab.Graphs
+using Catlab.CategoricalAlgebra
+using Catlab.CategoricalAlgebra.DataMigration
+using Test
+
+import Catlab.Theories: compose
+
+#=
+# Definition of a Preorder formalized as a GAT
```

Comment:
We should note that all these theories are defined in the Catlab standard library (`Catlab.Theories`).